=== modified file 'template/en/default/bug/create/create.html.tmpl'
--- template/en/default/bug/create/create.html.tmpl	2011-08-16 00:47:18 +0000
+++ template/en/default/bug/create/create.html.tmpl	2011-10-05 15:16:19 +0000
@@ -41,7 +41,7 @@
 
 <script type="text/javascript">
 <!--
-
+[%+ INCLUDE "assigneelist/bugcreate.js.tmpl" %]
 var initialowners = new Array([% product.components.size %]);
 var last_initialowner;
 var initialccs = new Array([% product.components.size %]);
@@ -102,6 +102,7 @@
     if (index != -1) {
         var owner = initialowners[index];
         var component = components[index];
+        populateAssignees(index);
         if (assigned_to == last_initialowner
             || assigned_to == owner
             || assigned_to == '') {
@@ -383,7 +384,7 @@
       field = bug_fields.assigned_to editable = 1
     %]
     <td colspan="2">
-      [% INCLUDE global/userselect.html.tmpl
+      [% INCLUDE assigneelist/userselect.html.tmpl
          id => "assigned_to"
          name => "assigned_to"
          value => assigned_to

=== modified file 'template/en/default/bug/edit.html.tmpl'
--- template/en/default/bug/edit.html.tmpl	2011-09-01 18:30:51 +0000
+++ template/en/default/bug/edit.html.tmpl	2011-10-05 15:18:35 +0000
@@ -32,6 +32,7 @@
 
   <script type="text/javascript">
   <!--
+[%+ INCLUDE "assigneelist/bugedit.js.tmpl" %]
 [% IF user.is_timetracker %]
   var fRemainingTime = [% bug.remaining_time %]; // holds the original value
   function adjustRemainingTime() {

